dollarskip.c

```diff
@@ -2,11 +2,103 @@
 #include <stdlib.h>
 #include <string.h>
 #include <assert.h>
+#include <stdbool.h>
 
 #ifndef COMMAND_MAX_SIZE
 #define COMMAND_MAX_SIZE 8096
 #endif
 
+/*** wrapped shell builtins ***/
+int easter_egg() {
+	printf("You found the DollarSkip easter egg!\n");
+	return 0;
+}
```

Comment:
Nevermind, I skipped past this part, i think this is pretty pointless, fun but pointless and dollarskip is made to merely pass the commands through, not be an actual application.

It should be a possibility that we remove this.

Reply:
I'm fine with removing it. I added it just as an example of howbto wrap shell commands that are important enough to do ourselves.

Reply:
We could also rename it to `dollarskip` and make it print about and version info. If that's what we do, I'll remove the overriding shell commands as we can check for arguments in main(). That will also improve speed as we remove one loop.
A faster way to do this would be with a hash table (`map` in go), but I didn't want to add my implementation (that is pretty big and not so readable).